src/Orleans.Streaming/MemoryStreams/MemoryStreamQueueGrain.cs

```diff
@@ -44,12 +45,25 @@ public Task<List<MemoryMessageData>> Dequeue(int maxCount)
         {
             List<MemoryMessageData> list = new List<MemoryMessageData>();
 
-            for (int i = 0; i < maxCount && eventQueue.Count > 0; ++i)
+            for (int i = 0; i < maxCount && _eventQueue.Count > 0; ++i)
             {
-                list.Add(eventQueue.Dequeue());
+                list.Add(_eventQueue.Dequeue());
             }
 
             return Task.FromResult(list);
         }
+
+        void IGrainMigrationParticipant.OnDehydrate(IDehydrationContext dehydrationContext)
+        {
+            dehydrationContext.TryAddValue("queue", _eventQueue);
```

Comment:
Not important right now: we should log when adding the _eventQueue failed, just to help debugging/catch some bugs